scripts/update_docker_images.sh

```diff
@@ -71,91 +65,72 @@ while getopts ":r:v:abx:h" opt; do
+image_names="base migrations common_tests foreman api_base api_production api_local \
+    transcriptome smasher salmon no_op illumina downloaders compendia"
+if [ "$BUILD_AFFYMETRIX" ]; then
+    image_names="$image_names affymetrix"
 fi
 
 # Set the version for the common project.
 echo "$SYSTEM_VERSION" >common/version
 
 # Create common/dist/data-refinery-common-*.tar.gz, which is
 # required by the workers and data_refinery_foreman images.
-## Remove old common distributions if they exist.
+# Remove old common distributions if they exist.
 rm -f common/dist/*
 (cd common && python3 setup.py sdist 1>/dev/null) # Run quietly in a subshell.
 
-ARCH="$(uname -m)"
-
-if [ "$ARCH" != "$SYSTEM_ARCH" ] && [ -z "$DOCKER_BUILDER" ]; then
-    echo "You're building $SYSTEM_ARCH images on the $ARCH machine without " \
-        "specifying a DOCKER_BUILDER. This may take a while..."
-    sleep 5
-fi
-
-for DOCKER_IMAGE in $DOCKER_IMAGES; do
-    case $DOCKER_IMAGE in
-    api)
-        DOCKER_FILE_PATH="api/dockerfiles/Dockerfile.api_production"
+# shellcheck disable=SC2086
+for image_name in $image_names; do
+    case $image_name in
+    api_base | api_local | api_production)
+        DOCKER_FILE_PATH="api/dockerfiles/Dockerfile.$image_name"
+        ;;
+    base | common_tests | migrations)
+        DOCKER_FILE_PATH="common/dockerfiles/Dockerfile.$image_name"
         ;;
     foreman)
-        DOCKER_FILE_PATH="foreman/dockerfiles/Dockerfile.$DOCKER_IMAGE"
+        DOCKER_FILE_PATH="foreman/dockerfiles/Dockerfile.$image_name"
         ;;
     *)
-        DOCKER_FILE_PATH="workers/dockerfiles/Dockerfile.$DOCKER_IMAGE"
+        DOCKER_FILE_PATH="workers/dockerfiles/Dockerfile.$image_name"
         ;;
     esac
 
-    IMAGE_NAME="$DOCKERHUB_REPO/dr_$DOCKER_IMAGE"
+    echo
+    echo "Building the $image_name:$SYSTEM_VERSION image from $DOCKER_FILE_PATH."
+    echo
 
-    echo "Building $SYSTEM_ARCH $IMAGE_NAME:$SYSTEM_VERSION image from " \
-        "$DOCKER_FILE_PATH."
+    DOCKERHUB_IMAGE="$DOCKERHUB_REPO/dr_$image_name"
+    CACHE_FROM_LATEST="cache-from=type=registry,ref=${DOCKERHUB_IMAGE}_cache:latest"
+    CACHE_FROM_VERSION="cache-from=type=registry,ref=${DOCKERHUB_IMAGE}_cache:$SYSTEM_VERSION"
+    CACHE_TO_LATEST="cache-to=type=registry,ref=${DOCKERHUB_IMAGE}_cache:latest,mode=max"
+    CACHE_TO_VERSION="cache-to=type=registry,ref=${DOCKERHUB_IMAGE}_cache:$SYSTEM_VERSION,mode=max"
 
-    if test "$DOCKER_BUILDER"; then
-        echo "Using builder $DOCKER_BUILDER."
-        docker buildx use "$DOCKER_BUILDER"
-    fi
+    set_up_docker_builder
 
-    DOCKER_BUILDKIT=1 docker buildx build \
-        --build-arg BUILDKIT_INLINE_CACHE=1 \
+    docker buildx build \
         --build-arg DOCKERHUB_REPO="$DOCKERHUB_REPO" \
         --build-arg SYSTEM_VERSION="$SYSTEM_VERSION" \
-        --cache-from "$IMAGE_NAME:$SYSTEM_VERSION" \
-        --cache-from "$IMAGE_NAME:latest" \
+        --"$CACHE_FROM_LATEST" \
```

Comment:
Do you think this looks more consistent with how we set the other values?
```suggestion
        cache-from="$CACHE_FROM_LATEST" \
```

Reply:
That's right, initially `cache-from/cache-to` options had more extras around them. It's no longer needed. I updated `scripts/update_docker_images.sh` too.